google-home-community.groovy

```diff
@@ -2091,6 +2438,29 @@ private deviceStateForTrait_Dock(deviceTrait, device) {
     ]
 }
 
+@SuppressWarnings('UnusedPrivateMethod')
+private deviceStateForTrait_EnergyStorage(deviceTrait, device) {
+    def deviceState = [
+        descriptiveCapacityRemaining: device.currentValue(deviceTrait.descriptiveCapacityRemainingAttribute)
+    ]
+    deviceState.capacityRemaining = [
+        [
+            rawValue: device.currentValue(deviceTrait.capacityRemainingRawValue).toInteger(),
+            unit:     deviceTrait.capacityRemainingUnit
+        ]
+    ]
+    deviceState.capacityUntilFull = [
+        [
+            rawValue: device.currentValue(deviceTrait.capacityUntilFullRawValue).toInteger(),
+            unit:     deviceTrait.capacityUntilFullUnit
+        ]
+    ]
+    deviceState.isCharging = device.currentValue(deviceTrait.isChargingAttribute) == deviceTrait.chargingValue
+    deviceState.isPluggedIn = device.currentValue(deviceTrait.isPluggedInAttribute) == deviceTrait.pluggedInValue
+
+    return deviceState
+}
```

Comment:
This method needs to be updated to gracefully handle when the optional attributes aren't set.  Currently it throws a bunch of `NullPointerException`s on sync.

Reply:
Sorry, missed that.  Rooky error and has been corrected.